src/Estimators/OneBodyDensityMatrices.h

```diff
@@ -0,0 +1,257 @@
+
+private:
+  OneBodyDensityMatricesInput input_;
+  Lattice lattice_;
+  SpeciesSet species_;
+  //data members
+  CompositeSPOSet basis_functions;
+  Vector<Value> basis_values;
+  Vector<Value> basis_norms;
+  Vector<Grad> basis_gradients;
+  Vector<Value> basis_laplacians;
+  Vector<Value> integrated_values;
+  bool warmed_up;
+  std::vector<Position> rsamples;
+  Vector<Real> sample_weights;
+  std::vector<Value> psi_ratios;
+  Real dens;
+  Position drift;
+  int nindex;
+  int eindex;
+  // TraceSample<TraceReal>* w_trace;
+  // TraceSample<TraceComp>* T_trace;
+  // CombinedTraceSample<TraceReal>* Vq_trace;
+  // CombinedTraceSample<TraceReal>* Vc_trace;
+  // CombinedTraceSample<TraceReal>* Vqq_trace;
+  // CombinedTraceSample<TraceReal>* Vqc_trace;
+
+  // CombinedTraceSample<TraceReal>* Vcc_trace;
+  std::vector<Value> E_samp;
+  // CombinedTraceSample<TraceReal>* E_trace;
+
+  bool initialized;
+  bool normalized;
+  bool volume_normed;
+  int basis_size;
+  int samples;
+  int nparticles;
+  int nspecies;
+  std::vector<int> species_size;
+  std::vector<std::string> species_name;
+  std::vector<Vector<Value>*> E_N;
+  std::vector<Matrix<Value>*> Phi_NB, Psi_NM, Phi_Psi_NB, N_BB, E_BB;
+  Matrix<Value> Phi_MB;
+  bool check_overlap;
+  bool check_derivatives;
+
+//#define DMCHECK
+#ifdef DMCHECK
+  std::vector<Vector<Value>*> E_Ntmp;
+  std::vector<Matrix<Value>*> Phi_NBtmp, Psi_NMtmp, Phi_Psi_NBtmp, N_BBtmp, E_BBtmp;
+  Matrix_t Phi_MBtmp;
+#endif
+
+  Position center, rcorner;
+  Real volume;
+  bool periodic;
+  int nmoves;
+  int naccepted;
+  Real acceptance_ratio;
+  bool write_acceptance_ratio;
+  bool write_rstats;
+
+  int ind_dims[OHMMS_DIM];
+  Real metric;
+
+  Position rpcur;
+  Position dpcur;
+  Real rhocur;
+
+  RandomGenerator_t* uniform_random;
+
+public:
+  /** Standard Constructor
+   *  If you are making a new OBDM this is what you should be calling
+   */
+  OneBodyDensityMatrices(OneBodyDensityMatricesInput&& obdmi, const Lattice& lattice, const SpeciesSet& species);
+
+  /** actual copy constructor
+   */
+  OneBodyDensityMatrices(const OneBodyDensityMatrices& obdm);
+  ~OneBodyDensityMatrices() override;
+
+  OneBodyDensityMatrices* clone() override;
+
+  void accumulate(const RefVector<MCPWalker>& walkers, const ParticleSet& pset_target, const RefVector<ParticleSet>& psets, const RefVector<TrialWaveFunction>& wfns, RandomGenerator_t& rng) override;
+
+  void normalize(Real invToWgt) override;
+  
+  void setRandomGenerator(RandomGenerator_t* rng);
+
+  void startBlock(int steps) override;
+
+  /** create and tie OperatorEstimator's observable_helper hdf5 wrapper to stat.h5 file
+   * @param gid hdf5 group to which the observables belong
+   *
+   * The default implementation does nothing. The derived classes which compute
+   * big data, e.g. density, should overwrite this function.
+   */
+  void registerOperatorEstimator(hid_t gid) override {}
+
+  // //standard interface
```

Comment:
Should remove all the commented out legacy functions.

Reply:
You're write I can still refer to DensityMatrices1B easily.